zeebe/expression-language/src/main/java/io/camunda/zeebe/el/impl/FeelEvaluationResult.java

```diff
@@ -0,0 +1,169 @@
+    this.result = result;
+    this.warnings = warnings;
+    this.messagePackTransformer = messagePackTransformer;
+  }
+
+  @Override
+  public String getExpression() {
+    return expression.getExpression();
+  }
+
+  @Override
+  public boolean isFailure() {
+    return false;
+  }
+
+  @Override
+  public String getFailureMessage() {
+    return null;
+  }
+
+  @Override
+  public List<EvaluationWarning> getWarnings() {
+    return warnings;
+  }
+
+  @Override
+  public ResultType getType() {
+    return switch (result) {
+      case final ValNull$ ignored -> ResultType.NULL;
+      case final ValBoolean ignored -> ResultType.BOOLEAN;
+      case final ValNumber ignored -> ResultType.NUMBER;
+      case final ValString ignored -> ResultType.STRING;
+      case final ValList ignored -> ResultType.ARRAY;
+      case final ValContext ignored -> ResultType.OBJECT;
+      case final ValDayTimeDuration ignored -> ResultType.DURATION;
+      case final ValYearMonthDuration ignored -> ResultType.PERIOD;
+      case final ValDate ignored -> ResultType.DATE;
+      case final ValDateTime ignored -> ResultType.DATE_TIME;
+      case final ValLocalDateTime ignored -> ResultType.DATE_TIME;
+      default -> ResultType.UNKNOWN;
+    };
+  }
+
+  @Override
+  public DirectBuffer toBuffer() {
+    return messagePackTransformer.apply(result);
+  }
+
+  @Override
+  public String getString() {
+    if (result instanceof final ValString s) {
+      return s.value();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public Boolean getBoolean() {
+    if (result instanceof final ValBoolean b) {
+      return b.value();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public Number getNumber() {
+    if (result instanceof final ValNumber n) {
+      return n.value();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public Duration getDuration() {
+    if (result instanceof final ValDayTimeDuration d) {
+      return d.value();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public Period getPeriod() {
+    if (result instanceof final ValYearMonthDuration d) {
+      return d.value();
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public ZonedDateTime getDateTime() {
+    return switch (result) {
+      case final ValDateTime dateTime -> dateTime.value();
+      case final ValLocalDateTime localDateTime ->
+          localDateTime.value().atZone(ZoneId.systemDefault());
+      default -> null;
+    };
+  }
+
+  @Override
+  public List<DirectBuffer> getList() {
+    if (result instanceof final ValList l) {
+      return asJava(l.items().map(v -> cloneBuffer(messagePackTransformer.apply(v))));
+    } else {
+      return null;
+    }
+  }
+
+  @Override
+  public List<String> getListOfStrings() {
+    if (result instanceof final ValList l && l.items().forall(ValString.class::isInstance)) {
+      return asJava(l.items().map(v -> ((ValString) v).value()));
```

Comment:
❓ Do we have test cases that verify the behavior for lists of mixed types? My Scala knowledge is not proficient enough to derive it from the original code but my understanding is that the original code would skip non-`ValString` elements while here we only return a list if all elements are `ValString`s. But I might be wrong with that.

Reply:
This is also confirmed by the only usage of `getListOfStrings` which has this error message:
>  "Expected result of the expression '%s' to be 'ARRAY' containing 'STRING' items, but was 'ARRAY' containing at least one non-'STRING' item."